### PR TITLE
Update tests

### PR DESCRIPTION
**Pull request by kgryte**
This PR moves files around to minimize the need to rebuild when updating tests and modifies some of the test commands.

### CODE REVIEW

File: jest-puppeteer.config.js [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
+ *
+ * Copyright (c) 2019 Project Jupyter Contributors.
+ * Distributed under the terms of the 3-Clause BSD License.
+ */
+
+const config = {
   launch: {
-    headless: process.env.HEADLESS !== "false"
+    headless: process.env.HEADLESS !== 'false',
+    slowMo: process.env.SLOWMO === 'true'
```
Comment (saulshanabrook) [inline jest-puppeteer.config.js:11]: Nice

File: jest-puppeteer.config.js [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
   },
   // https://github.com/smooth-code/jest-puppeteer/tree/master/packages/jest-dev-server#options
   server: {
-    command: "jupyter lab --port 8080 --no-browser",
+    command: "jupyter lab --port 8080 --no-browser --LabApp.token=''",
```
Comment (saulshanabrook) [inline jest-puppeteer.config.js:15]: Ah good call

Comment (kgryte) [reply jest-puppeteer.config.js:15]: One less configuration file.

File: tsconfig.test.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "outDir": "build/test",
+    "rootDir": "test"
+  },
+  "include": ["test/*", "test/**/*"]
```
Comment (saulshanabrook) [inline tsconfig.test.json:7]: Yeah probably good call to separate these

File: jest-puppeteer.config.js (hunk 1 above)
Comment (kgryte) [reply jest-puppeteer.config.js:11]: Was definitely useful in seeing what button was being pressed in real-time.

File: test/ui/test.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+function sleep(ms: number): Promise<void> {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+describe('JupyterLab', () => {
+  beforeAll(async () => {
+    // Load JupyterLab:
+    await page.goto('http://localhost:8080/lab?reset');
+
+    // NOTE: depending on system resource constraints, this may NOT be enough time for JupyterLab to load and get "settled", so to speak. If CI tests begin inexplicably failing due to timeout failures, may want to consider increasing the sleep duration...
```
Comment (saulshanabrook) [inline test/ui/test.ts:32]: :D

File: jest-environment.js [hunk 5]
```diff
@@ -1,17 +1,26 @@
...
+ *
+ * Copyright (c) 2019 Project Jupyter Contributors.
+ * Distributed under the terms of the 3-Clause BSD License.
+ */
+
 // Based on from https://yarnpkg.com/en/package/@rws-air/jestscreenshot
 
-const PuppeteerEnvironment = require("jest-environment-puppeteer");
-const JestScreenshot = require("@rws-air/jestscreenshot");
-require("jest-circus");
```
Comment (saulshanabrook) [inline jest-environment.js:5]: It's interesting you didn't need this! Not sure why we did, they just had it here.

**Comment by saulshanabrook** [comment]
I'm loving the green checkmark

File: tsconfig.test.json (hunk 3 above)
Comment (kgryte) [reply tsconfig.test.json:7]: I recognize that, in other worlds (e.g., Go, Ruby), placing tests next to source files is common; in Node.js land, it isn't for reasons of separation of concerns. Plus, UI tests cross file divides and don't fit nicely into the `foo.spec.ts` paradigm where `foo` refers to `foo.ts`, a source file.

File: jest-environment.js (hunk 5 above)
Comment (kgryte) [reply jest-environment.js:5]: This was present in one of the dep documentation examples. I am not sure why it was there. If it was needed, I would disappointed, as it means importing the package has (gasp) side-effects :|.

**Pull request merged**